### PR TITLE
add append/prepend tx message instruction

### PR DESCRIPTION
**Pull request by mcintyre94**
Implementation + tests copied from @solana/transactions

### CODE REVIEW

**Comment by mcintyre94** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2429?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2429** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2429?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2428** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2428?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2427** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2427?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2426** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2426?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2425** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2425?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2424** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2424?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2423** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2423?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mcintyre94 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/transaction-messages/src/instructions.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { BaseTransactionMessage } from './transaction-message';
+
+// TODO remove
+interface ITransactionWithSignatures {}
```
Comment (steveluscher) [inline packages/transaction-messages/src/instructions.ts:4]: Now?

**Review by steveluscher — APPROVED** [review]
Added via Giphy

File: packages/transaction-messages/src/instructions.ts (hunk 1 above)
Comment (mcintyre94) [reply packages/transaction-messages/src/instructions.ts:4]: Before committing ideally 😅

**Pull request merged**